server.js

```diff
@@ -1,15 +1,27 @@
 'use strict';
-var crypto = require('crypto');
-var Q = require('q');
+
+/* Libraries */
+var compression = require('compression');
 var connect = require('connect');
-var morgan = require('morgan');
+var crypto = require('crypto');
 var errorhandler = require('errorhandler');
+var morgan = require('morgan');
+var yeomanPluginCache = require('yeoman-plugin-cache');
 var timeout = require('connect-timeout');
-var compression = require('compression');
-var pluginCache = require('./');
 
-var HTTP_PORT = process.env.PORT || 8001;
 
+/* env Variables */
```

Comment:
I think this block of code is trying to be very clever but end up being quite confusing.

We don't need to make every parameter optional, and we don't need to make smart decision when they're not defined.

That being said, we can also simplify that logic by creating a function taking an object of parameter and using `_.extend()` to define defaults as a clear list of values. This would just clarify our intents and the default definition somewhere where it is easily readable.